deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -259,6 +278,29 @@ boolean canDeploy(EnvironBean envBean, String host, AgentBean agentBean, String
         }
     }
 
+    boolean canDeployWithRule(String hostId, EnvironBean envBean) throws Exception{
+        String envId = envBean.getEnv_id();
+        String ruleId = envBean.getDeploy_rule_id();
+
+        try {
+            LOG.debug("verify active agents for host {} rule {}", hostId, ruleId);
+            DeployRuleBean deployRuleBean = deployRuleDAO.getById(ruleId);
+            String tagValue = deployRuleBean.getCondition_key();
+            String tagName = deployRuleBean.getCondition_value();
+
+            long maxParallelWithHostTag = deployRuleBean.getMax_parallel();
+            long totalActiveAgentsWithHostTag = agentDAO.countDeployingAgentWithHostTag(envBean.getEnv_id(), tagName, tagValue);
+            if(totalActiveAgentsWithHostTag >= maxParallelWithHostTag) {
+                LOG.info("already exceed {} for env = {}, host = {}, rule = {}, return false", maxParallelWithHostTag, envId, hostId, deployRuleBean.toString());
```

Comment:
seems that this line is to long. I also notice that there is no style check in pom file, and maybe we should add that (not necessary in this PR) :-)